p2p/src/test/kotlin/maru/p2p/MaruPeerManagerTest.kt

```diff
@@ -44,164 +63,171 @@ class MaruPeerManagerTest {
-        isStaticPeer = { false },
-      )
+    private fun createSyncStatusProvider(): SyncStatusProvider =
+      object : SyncStatusProvider {
+        override fun getCLSyncStatus(): CLSyncStatus = CLSyncStatus.SYNCED
 
-    // Act
-    manager.start(discoveryService = null, p2pNetwork = mock())
-    manager.onConnect(peer)
+        override fun getElSyncStatus(): ELSyncStatus = ELSyncStatus.SYNCED
 
-    // Assert
-    verify(maruPeer).sendStatus()
-    verify(maruPeer, never()).scheduleDisconnectIfStatusNotReceived(any())
-  }
+        override fun onClSyncStatusUpdate(handler: (newStatus: CLSyncStatus) -> Unit) {}
 
-  @Test
-  fun `sends status message immediately for locally initiated connections`() {
-    // Arrange
-    val nodeId: NodeId = mock()
-    val peer = Fixtures.tekuPeer(id = nodeId, initiatedLocally = true)
-    val maruPeerFactory = mock<MaruPeerFactory>()
-    val mockFutureStatus = mock<SafeFuture<Unit>>()
-    val maruPeer = Fixtures.maruPeer(initiatedLocally = true, sendStatusFuture = mockFutureStatus)
-    val p2pConfig = P2PConfig()
+        override fun onElSyncStatusUpdate(handler: (newStatus: ELSyncStatus) -> Unit) {}
 
-    whenever(maruPeerFactory.createMaruPeer(peer)).thenReturn(maruPeer)
+        override fun isBeaconChainSynced(): Boolean = true
 
-    val manager =
-      MaruPeerManager(
-        maruPeerFactory = maruPeerFactory,
-        p2pConfig = p2pConfig,
-        reputationManager = reputationManager,
-        isStaticPeer = { false },
-      )
+        override fun isELSynced(): Boolean = true
 
-    // Act
-    manager.start(discoveryService = null, p2pNetwork = mock())
-    manager.onConnect(peer)
+        override fun onBeaconSyncComplete(handler: () -> Unit) {}
 
-    // Assert
-    verify(maruPeer).sendStatus()
+        override fun onFullSyncComplete(handler: () -> Unit) {}
+
+        override fun getBeaconSyncDistance(): ULong = 0UL
+
+        override fun getCLSyncTarget(): ULong = 0UL
+      }
+
+    val syncStatusProvider: SyncStatusProvider = createSyncStatusProvider()
   }
 
   @Test
-  fun `does not send status message for remotely initiated connections`() {
-    // Arrange
-    val nodeId: NodeId = mock()
-    val address = Fixtures.peerAddress(nodeId)
-    val peer = Fixtures.tekuPeer(id = nodeId, address = address, initiatedLocally = false)
+  fun `does not schedule timeout when connection is initiated locally`() {
+    val peer = Fixtures.tekuPeer(id = mock(), initiatedLocally = true)
     val maruPeerFactory = mock<MaruPeerFactory>()
-    val maruPeer = Fixtures.maruPeer(initiatedLocally = false, address = address)
-    val p2pConfig = P2PConfig()
+    val maruPeer = Fixtures.maruPeer()
+    val p2pConfig = mock<P2PConfig>()
+    val syncConfig = mock<SyncingConfig>()
 
-    whenever(maruPeer.getStatus()).thenReturn(null)
     whenever(maruPeerFactory.createMaruPeer(peer)).thenReturn(maruPeer)
+    whenever(p2pConfig.maxPeers).thenReturn(10)
```

Comment:
Suggestion:

You're mocking this config because `MaruPeerManager` doesn't use the whole P2PConfig. So what you can do is to introduce `MaruPeerManager.Config` which only has the relevant 2 fields. And it will be instantiated using `p2pConfig` where it's instantiated